src/client/mod.rs

```diff
@@ -356,16 +417,20 @@ impl WarpgrapherClient {
     /// use std::env::var_os;
```

Comment:
The documentation just above should describe conditions under which errors are returned.